includes/handle.inc

```diff
@@ -27,13 +27,26 @@ function islandora_handle_construct_handle(AbstractObject $object) {
   ));
   $query_target = drupal_http_build_query(array('target' => $target));
   $authorization_header = 'Basic ' . base64_encode($admin_user . ':' . $admin_password);
-  $response = drupal_http_request($handle_service_url, array(
-    'headers' => array(
-      'Authorization' => $authorization_header,
-    ),
-    'data' => $query_target,
-    'method' => 'POST',
-  ));
+  if (variable_get('islandora_handle_debug_mode', FALSE)) {
+    $response = new stdClass();
+    $response->code = 201;
+    db_insert('islandora_handle_debug_handles')
+      ->fields(array(
+        'suffix' => $suffix,
+        'handle' => $handle_service_url,
+      ))
+      ->execute();
+    return $response;
```

Comment:
With the else the return isn't necessary and visa-versa; just one would be better.